tests/flytekit/unit/core/test_type_engine.py

```diff
@@ -1404,6 +1405,65 @@ class Bar(DataClassJsonMixin):
         DataclassTransformer().assert_type(gt, pv)
 
 
+def test_assert_dict_type():
+    @dataclass
+    class Args(DataClassJsonMixin):
+        x: int
+        y: typing.Optional[str]
```

Comment:
1. if `y` is optional, we should give it a default value.
2. could we also add `flytefile`, `StructuredDataset` and `dataclass` to this class?
```suggestion
        y: typing.Optional[str] = None
        f: FlyteFile
        s: StructuredDataset
        d: AnotherDataclass
```

Reply:
Thanks for the suggestion, I've added it.